xdsl/dialects/riscv_scf.py

```diff
@@ -40,13 +40,10 @@
 class YieldOp(AbstractYieldOperation[RISCVRegisterType]):
     name = "riscv_scf.yield"
 
-    traits = traits_def(lambda: frozenset([IsTerminator(), HasParent(ForOp, WhileOp)]))
+    traits = traits_def(lambda: frozenset([IsTerminator(), HasParent(WhileOp, ForRofBaseOp)]))
 
 
-@irdl_op_definition
-class ForOp(IRDLOperation):
-    name = "riscv_scf.for"
-
+class ForRofBaseOp(IRDLOperation):
```

Comment:
```suggestion
class ForRofOperation(IRDLOperation, abc.ABC):
```
We usually use Operation for abstract superclasses